services/tables/src/main/java/com/linkedin/openhouse/tables/utils/CronScheduleGenerator.java

```diff
@@ -0,0 +1,72 @@
+package com.linkedin.openhouse.tables.utils;
+
+import com.cronutils.builder.CronBuilder;
+import com.cronutils.model.CronType;
+import com.cronutils.model.definition.CronDefinitionBuilder;
+import com.cronutils.model.field.expression.FieldExpression;
+import com.cronutils.model.field.expression.FieldExpressionFactory;
+import com.linkedin.openhouse.common.exception.RequestValidationFailureException;
+import com.linkedin.openhouse.tables.api.spec.v0.request.components.ReplicationConfig;
+import java.util.Random;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Component;
+
+/** Utility class for generating a cron schedule given an interval for which a job should run */
+@Slf4j
+@Component
+public class CronScheduleGenerator {
+
+  /**
+   * Public api to generate a cron schedule for a {@link ReplicationConfig} based on a given
+   * interval string in the form 12H, 1D, 2D, 3D.
+   *
+   * @param interval
+   * @return schedule
+   */
+  public static String buildCronExpression(String interval) {
+    if (interval == null || interval.isEmpty()) {
+      log.error("Replication interval is null or empty");
+      throw new RequestValidationFailureException(
+          String.format("Replication interval is null or empty"));
```

Comment:
Extract error msg in a variable will avoid duplication.

Reply:
thanks, updated 👍